### PR TITLE
Changea ExtractedZipFile inputstream to supplier

### PR DESCRIPTION
**Pull request by meg1502**
To be able to process these files multiple times as it often is done for xml files during import jobs

### CODE REVIEW

File: src/main/java/sirius/biz/util/ArchiveExtractor.java [hunk 1]
```diff
@@ -257,9 +257,16 @@ private void extractZipEntriesFromZipFile(Predicate<String> filter,
...
             if (!entry.isDirectory() && filter.test(entry.getName())) {
                 Amount progress = Amount.of(numberOfFiles).divideBy(Amount.of(zipFile.size()));
-                boolean shouldContinue = extractedFileConsumer.apply(new ExtractedZipFile(entry,
-                                                                                          zipFile.getInputStream(entry),
-                                                                                          progress));
+                boolean shouldContinue = extractedFileConsumer.apply(new ExtractedZipFile(entry, () -> {
+                    try {
+                        return zipFile.getInputStream(entry);
+                    } catch (IOException e) {
+                        throw Exceptions.createHandled()
```
Comment (andyHa) [inline src/main/java/sirius/biz/util/ArchiveExtractor.java:264]: i'd remove this error handling here? (as it presumes createHandled)

File: src/main/java/sirius/biz/util/ExtractedZipFile.java [hunk 2]
```diff
@@ -25,18 +24,18 @@
 class ExtractedZipFile implements ExtractedFile {
 
     private final ZipEntry entry;
-    private final InputStream zipInputStream;
+    private final Supplier<InputStream> inputStreamSupplier;
     private final Amount progress;
 
-    ExtractedZipFile(ZipEntry entry, InputStream zipInputStream, Amount progress) {
+    ExtractedZipFile(ZipEntry entry, Supplier<InputStream> inputStreamSupplier, Amount progress) {
```
Comment (andyHa) [inline src/main/java/sirius/biz/util/ExtractedZipFile.java:30]: use a Provider here...

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/biz/util/ExtractedZipFile.java [hunk 3]
```diff
@@ -25,18 +24,18 @@
...
         this.entry = entry;
-        this.zipInputStream = zipInputStream;
+        this.inputStreamSupplier = inputStreamSupplier;
         this.progress = progress;
     }
 
     @Override
     public InputStream openInputStream() throws IOException {
-        return new CloseShieldInputStream(zipInputStream);
+        return new CloseShieldInputStream(inputStreamSupplier.get());
```
Comment (andyHa) [inline src/main/java/sirius/biz/util/ExtractedZipFile.java:38]: catch and rethrow IOException catch and properly handle all other exceptions as .handled().

**Review by andyHa — CHANGES_REQUESTED** [review]
...

**Review by andyHa — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Pull request merged**